### PR TITLE
Initial commit for Streamline LifterLMS

### PR DESCRIPTION
**Pull request by MaximilianoRicoTabo**
image

image
image

 * Create the includes/compatibility/lifterlms.php file.
 * Create a prompt if LifterLMS is activated for the first time when PMPro is already installed. It should show up before the Lifter wizard and *look* like the Lifter Wizard.
 * Add option to enable/disable the streamlined mode

### All Submissions:

* [x] Have you followed the [Contributing guideline](https://github.com/strangerstudios/paid-memberships-pro/blob/master/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/strangerstudios/paid-memberships-pro/pulls) for the same update/change?

### Changes proposed in this Pull Request:

Resolves XXX.

### How to test the changes in this Pull Request:

1.
2.
3.

### Other information:

* [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you successfully run tests with your changes locally?

### Changelog entry

> Enter a summary of all changes on this Pull Request. This will appear in the changelog if accepted.

### CODE REVIEW

File: includes/compatibility/lifter/lifter.php [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+add_action( 'activated_plugin', 'wp_detect_plugin_activation', 10, 2 );
+
+function toggle_streamline() {
+  $status = "true";  $_POST['status'];
+  update_option( 'toggle_streamline', $status);
+  exit();
+}
+
+add_action( 'wp_ajax_toggle_streamline', 'toggle_streamline' );
```
Comment (MaximilianoRicoTabo) [inline includes/compatibility/lifter/lifter.php:36]: All this need to be documented

File: includes/functions.php [hunk 2]
```diff
@@ -101,7 +101,7 @@ function pmpro_setOption( $s, $v = null, $sanitize_function = 'sanitize_text_fie
 	if ( is_array( $v ) ) {
 		$v = implode( ',', $v );
 	} else {
-		$v = trim( $v );
+		$v =isset ( $v ) ? trim( $v ) : '';
```
Comment (MaximilianoRicoTabo) [inline includes/functions.php:104]: this fixes a warning due  when $v is null

File: includes/adminpages.php [hunk 3]
```diff
@@ -69,6 +69,11 @@ function pmpro_add_pages() {
 	}
 	add_submenu_page( 'pmpro-dashboard', __( 'License', 'paid-memberships-pro' ), __( '<span style="color: ' . $span_color . '">License</span>', 'paid-memberships-pro' ), 'manage_options', 'pmpro-license', 'pmpro_license_settings_page' );
 
+
+	add_submenu_page( 'pmpro-members', __( 'Add Member', 'paid-memberships-pro' ), __( '<span>Add Member</span>', 'paid-memberships-pro' ), 'manage_options', 'pmpro-members', 'pmpro_members' );
```
Comment (ideadude) [inline includes/adminpages.php:73]:
```suggestion
```

Comment (ideadude) [reply includes/adminpages.php:73]: I think this slipped in from some of the other single member page work.

File: includes/compatibility/lifter/lifter.php [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+/**
+ * On LifterLMS plugin activation, activate streamline feature and install PMPro Courses plugin.
+ */
+function lifter_plugin_activation() {
+	update_option( 'pmpro_toggle_lifter_streamline_setup', 'true' );
+	include_once( PMPRO_DIR . '/classes/' . 'class-quiet-plugin-installer.php' ) ;
+	$slug = 'pmpro-courses';
+	$installer = new Quiet_Skin(compact('title', 'url', 'nonce', 'plugin', 'api') );
+	$installer->download_install_and_activate($slug);
+	exit( wp_redirect("/wp-admin/admin.php?page=pmpro-lifter-streamline") );
```
Comment (ideadude) [inline includes/compatibility/lifter/lifter.php:28]: In my testing, this exits before Lifter gets fully activated. Maybe intentional?

**Review by ideadude — COMMENTED** [review]
Some notes for me. Not intended for you to tackle on your own @MaximilianoRicoTabo. I will be looking into this more and we should plan to pair program through some of these next week together.

File: adminpages/lifter-streamline.php [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+<p><?php esc_html_e( 'It will only take a few minutes and it is completely optional. If you don\'t have the time now, come back later.', 'paid-memberships-pro' )  ?></p>
+<p><?php esc_html_e( 'Since you already have Paid Memberships Pro installed, you can enable a "streamlined" version of LifterLMS that will let PMPro handle all checkouts, memberships, restrictions, and user fields.', 'paid-memberships-pro' ) ?> </p>
+
+<label for="lifter-streamline">
+    <input type="checkbox" name="lifter-streamline" id="lifter-streamline" checked="checked">
+    <?php esc_html_e( 'Enable streamlined version of LifterLMS', 'paid-memberships-pro' ) ?>
+</label>
+
+        <p class="llms-setup-actions">
+            <a href="http://wp-core.dvl.to/wp-admin/" class="llms-button-secondary large"><?php esc_html_e( 'Skip setup' ) ?></a>
```
[the remark this answers is not in the document]
Comment (MaximilianoRicoTabo) [reply adminpages/lifter-streamline.php:57]: i think i changed all these link yesterday with this commit https://github.com/strangerstudios/paid-memberships-pro/pull/2518/commits/03a7d24cce3ae41437eac4392c8a63abdfbeb158, perhaps comments where made before ?  🤔

File: adminpages/advancedsettings.php [hunk 6]
```diff
@@ -53,7 +53,9 @@
 		pmpro_setOption("wisdom_opt_out");
 		pmpro_setOption("hideadslevels");
 		pmpro_setOption("redirecttosubscription");
-		pmpro_setOption("uninstall");		
+		pmpro_setOption("uninstall");
+		$isChecked = isset( $_POST['lifter-streamline'] ) ? 'true' : 'false';
```
[the remark this answers is not in the document]
Comment (ideadude) [reply adminpages/advancedsettings.php:57]: One alternative is to use this filter on the advanced settings page to save the field: https://github.com/strangerstudios/paid-memberships-pro/blob/dev/adminpages/advancedsettings.php#L64-L77

And here to show it:
https://github.com/strangerstudios/paid-memberships-pro/blob/dev/adminpages/advancedsettings.php#L427

File: css/admin.css [hunk 7]
```diff
@@ -2367,4 +2367,158 @@ button.pmpro_report_th_closed:before {
 
 .pmpro_plugin_update_notice+p:before {
 	display: none;
+}
```
[the remark this answers is not in the document]
Comment (ideadude) [reply css/admin.css:2370]: Even though we removed some of the markup and stuff, I think we still need to remove this CSS.

**Comment by dparker1005** [comment]
Closing in favor of #2533 which is built off of this work.

**Pull request closed without merging**